clang/lib/Basic/Cuda.cpp

```diff
@@ -96,6 +96,7 @@ static const OffloadArchToStringMap arch_names[] = {
     SM(89),                          // Ada Lovelace
     SM(90),                          // Hopper
     SM(90a),                         // Hopper
+    SM(next),                        // Placeholder for a new arch.
```

Comment:
I'm guessing it's intentional that when we add a new `sm` it will take that old value that `sm_next` had?

Reply:
The position of the entry in the table does not matter. Each entry will contain appropriate `OffloadArch::SM_##sm` enum. `SM(next)` will carry its own `OffloadArch::SM_next`. and it will remain so even when a new GPU gets added. The new GPU will get its own enum.

Reply:
Okay, so we don't rely on the actual integers here and don't need to worry about weird ABI stuff I guess.